### PR TITLE
HIVE-26444: Use tez.local.mode.without.network for unit tests

### PR DESCRIPTION
**Pull request by abstractdog**
### What changes were proposed in this pull request?
Described on jira.

### Why are the changes needed?
Test stability.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Full precommit testing.

### CODE REVIEW

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java [hunk 1]
```diff
@@ -201,6 +201,8 @@ private HiveConf initializeConf() {
 
     // Tez configuration
     hiveConf.setBoolean("tez.local.mode", true);
+    // TODO: enable below option once HIVE-26445 is investigated
+    // hiveConf.setBoolean("tez.local.mode.without.network", true);
```
Comment (zabetak) [inline iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java:205]: If there are failures due to the change then it becomes a bit doubtful if setting the property always to true is a good idea.

**Review by zabetak — APPROVED** [review]
I don't know much about `tez.local.mode.without.network` property but the current changeset looks reasonable. Since @abstractdog added this new property in Tez, I trust their judgement that it will improve test stability.

As I commented under TEZ-4236, I think it would make more sense if we didn't have to set the tez.local.mode.without.network explicitly but maybe this deserves a separate discussion.

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java (hunk 1 above)
Comment (ayushtkn) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java:205]: Not sure but it says in the description:
```
which is in the same JVM as the client.
```
May be some instance is getting shared by the client and one change is impacting other, hence the NPE which is mentioned in HIVE-26445.

I share the same doubt as @zabetak , we should atleast figure out the reason for failure if it is nothing core-related this property, then this should be ok.

**Review by ayushtkn — APPROVED** [review]
The change LGTM. not very sure about the working of ``tez.local.mode.without.network`` but I trust @abstractdog on that :-)

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java (hunk 1 above)
Comment (abstractdog) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java:205]: recently, only hive iceberg related tests failed [here](http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-3491/2/pipeline), and I cannot easily disable this feature only for TestIcebergLlapLocalCliDriver (but not for other CliDrivers), so I'm about to make this change only for some unit tests (which I targeted originally), and handle all the qtests in the followup HIVE-26445, does it make sense?

Comment (zabetak) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java:205]: Sounds good.

**Pull request merged**